### PR TITLE
implement the AFL-Style Tui

### PR DESCRIPTION
**Pull request by ToSeven**

### CODE REVIEW

**Comment by domenukk** [comment]
Can you fix the hints from clippy (./scripts/clippy.sh)?

**Comment by domenukk** [comment]
Does the UI still look okay if AflStatsStage isn't used?

File: libafl/src/monitors/tui/mod.rs [hunk 1]
```diff
@@ -178,7 +218,46 @@ impl ClientTuiContext {
...
+            "pending": 0,
+            "pend_fav": 0,
+            "imported": 0,
+            "own_finds": 0,
+        });
+        let afl_stats = client
+            .get_user_stats("AflStas")
+            .map_or(default_json.to_string(), ToString::to_string);
+
+        let afl_stats_json: Value = serde_json::from_str(afl_stats.as_str()).unwrap();
```
Comment (tokatoka) [inline libafl/src/monitors/tui/mod.rs:251]: please try not to use unwrap() in lib
use return Result<(), Error> in this function

File: libafl/src/monitors/tui/mod.rs [hunk 2]
```diff
@@ -178,7 +218,46 @@ impl ClientTuiContext {
...
+            .map_or("0%".to_string(), ToString::to_string);
+
+        let default_json = serde_json::json!({
+            "pending": 0,
+            "pend_fav": 0,
+            "imported": 0,
+            "own_finds": 0,
+        });
+        let afl_stats = client
+            .get_user_stats("AflStas")
```
Comment (tokatoka) [inline libafl/src/monitors/tui/mod.rs:248]: typo?

**Comment by ToSeven** [comment]
> Does the UI still look okay if AflStatsStage isn't used?

Yeah, I have tested the UI under this situation, all is Okay. But I donn't validate the UI if AflStatsStage is used.

File: libafl/src/monitors/tui/mod.rs (hunk 2 above)
Comment (ToSeven) [reply libafl/src/monitors/tui/mod.rs:248]: yeah, 🤦🏻‍♀️

**Comment by novafacing** [comment]
To update this PR branch to use ratatui, all you need to do is merge main into it (just checked locally).

**Pull request merged**